src/state_point.F90

```diff
@@ -238,59 +276,108 @@ subroutine write_state_point()
             call close_group(filter_group)
           end do FILTER_LOOP
 
-          call write_dataset(tally_group, "n_nuclides", tally%n_nuclide_bins)
-
           ! Set up nuclide bin array and then write
-          allocate(key_array(tally%n_nuclide_bins))
+          allocate(str_array(tally%n_nuclide_bins))
           NUCLIDE_LOOP: do j = 1, tally%n_nuclide_bins
             if (tally%nuclide_bins(j) > 0) then
-              key_array(j) = nuclides(tally%nuclide_bins(j))%zaid
+              i_list = nuclides(tally%nuclide_bins(j))%listing
+              i_xs = index(xs_listings(i_list)%alias, '.')
+              if (i_xs > 0) then
+                str_array(j) = xs_listings(i_list)%alias(1:i_xs - 1)
+              else
+                str_array(j) = xs_listings(i_list)%alias
+              end if
             else
-              key_array(j) = tally%nuclide_bins(j)
+              str_array(j) = 'total'
             end if
           end do NUCLIDE_LOOP
-          call write_dataset(tally_group, "nuclides", key_array)
-          deallocate(key_array)
+          call write_dataset(tally_group, "nuclides", str_array)
+          deallocate(str_array)
 
           call write_dataset(tally_group, "n_score_bins", tally%n_score_bins)
-          call write_dataset(tally_group, "score_bins", tally%score_bins)
+          allocate(str_array(size(tally%score_bins)))
+          do j = 1, size(tally%score_bins)
```

Comment:
Again, I like the idea of a dictionary mapping score enumerated types to string definitions, but it is up to you.